Sources/AnyPromise.h

```diff
@@ -2,15 +2,35 @@
 #import <dispatch/dispatch.h>
 #import <PromiseKit/fwd.h>
 
-/// INTERNAL DO NOT USE
-@class __AnyPromise;
+// Different ways to make AnyPromise available
+//
+// Import the Swift-Module if it was created already
+//
+// If building a Swift-Package just import PromiseKit
+//
+// In all other cases AnyPromise is still just a foward
+// declaration and needs to be defined. This is just to
+// silence a compile error which would avoid building
+// the Swift-Module on first builds.
+
+#if __has_include("PromiseKit-Swift.h")
```

Comment:
The Problem here was, that AnyPromise was not always defined at this point. But it must be defined here, because otherwise I could not declare a category on it. The include path is always different depending on what is currently build. Tests, Package, Xcode Project, etc. So I figured out that this seemed to work. I'm not very satisfied with it but finally it worked. Would appreciate a better solution. 

`self->d` isn't used anymore.  The idea I was following was defining `AnyPromise` in Swift and declaring a category in ObjC on that and delegate from the category to the Swift implementation.